aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -92,12 +92,19 @@ static void reciprocal_kernel(TensorIterator& iter) {
 }
 
 static void neg_kernel(TensorIterator& iter) {
-  AT_DISPATCH_ALL_TYPES(iter.dtype(), "neg_cpu", [&]() {
+  if (iter.dtype() == ScalarType::Bool) {
```

Comment:
This doesn't look right.  We shouldn't implement unary `-` on bool. We should implement bitwise not (unary `~`) instead.

It's also worth noting that this is the opposite of C++ behavior. For example:

```
bool x = ...
bool y = -x;
```

x and y have the same value (because 0=>0=>false and 1=>-1=>true)

Reply:
removing this change for now. will implement this as a separate PR